crates/shared/src/order_validation.rs

```diff
@@ -246,8 +247,14 @@ impl OrderValidator {
             quoter,
             balance_fetcher,
             signature_validator,
+            enable_limit_orders: false,
         }
     }
+
+    pub fn with_limit_orders(mut self, enable: bool) -> Self {
+        self.enable_limit_orders = enable;
+        self
+    }
```

Comment:
nit: Seems inconsistent to have this one argument be set through builder style while the rest stay in new. I'd add it to the other arguments and if we want to change the initialization style we can discuss it separately.

Reply:
This is only temporary. It was kind of convenient to do it this way for tests. But I think you're right.

Reply:
On second thought. I disagree. This is a boolean parameter. With boolean parameters are particularly tricky to see what is going on:
```rs
some_call(true); // True what?
```
So I'd still keep this as a setter function.